mkosi

```diff
@@ -791,9 +894,21 @@ def set_root_password(args, workspace):
                            if line.startswith('root:') else line)
         patch_file(os.path.join(workspace, 'root', 'etc/shadow'), jj)
 
-def install_boot_loader_arch(args, workspace):
+def install_boot_loader_arch(args, workspace, loopdev):
+    if args.luks_encryption == 'all':
+        c = subprocess.run(["lsblk", "-n", "--output", "UUID", partition(loopdev, args.root_partno)],
+                       stdout=subprocess.PIPE, check=True)
+        root_uuid = c.stdout.decode("utf-8").split()[0]
+        args.kernel_commandline += " luks.uuid={} luks.name={}=root root=/dev/mapper/root".format(root_uuid, root_uuid)
+        with open(os.path.join(workspace, "root", "etc/kernel/cmdline"), "w") as cmdline:
+            cmdline.write(args.kernel_commandline)
+            cmdline.write("\n")
+        hooks = "HOOKS=\"systemd modconf block keyboard sd-encrypt filesystems fsck\"\n"
```

Comment:
hmmm, i am currently working on making gpt auto discovery work in systemd for encrypted partitions. I'd really prefer if we could go that route in the long run, so that we don't have to define anything here... but for now, doing it this way is fine I figure.

I think instead of reading the UUID from the root partition I think it would make sense to set it to some chosen value right-away and then reusing that. i.e. use `--uuid=` on the luksFormat command line to pick something, and then just reuse it here. That means we don't have to shell out lsblk here, which I think it's a good thing.